File: 2.1. Two Pointers/2. Two Sum II Input Array Is Sorted/dory.py
# Runtime 108ms
# Beats 37.35%

# Memory 17.81MB
# Beats 5.48%
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        dic = {}
        for key, value in enumerate(numbers):
            buffer = target - value
            # print(buffer)
            if value in dic:
                return [dic[value]+1, key+1]
            else:
                dic[buffer] = key

File: 1. Arrays & Hashing/2. Valid Anagram/dory.cs
// Runtime 31ms
// Beats 24.36%

// Memory 44.72MB
// Beats 12.57%

public class Solution {
    public bool IsAnagram(string s, string t) {
        if(s.Length != t.Length) {
            return false;
        }
        char[] s_char = s.ToCharArray();
        char[] t_char = t.ToCharArray();
        Array.Sort(s_char);
        Array.Sort(t_char);
        return new string(s_char) == new string(t_char);
    }
}